js/src/components/contact-information/phone-number-card/phone-number-card.js

```diff
@@ -117,14 +117,27 @@ const PhoneNumberCard = ( {
 } ) => {
 	const { loaded, data } = phoneNumber;
 	const [ isEditing, setEditing ] = useState( initEditing );
+	const onPhoneNumberVerifiedRef = useRef();
+	onPhoneNumberVerifiedRef.current = onPhoneNumberVerified;
+
+	const { isVerified } = data;
 
 	// Handle the initial UI state of `initEditing = null`.
 	// The `isEditing` state is on hold. Determine it after the `phoneNumber` loaded.
 	useEffect( () => {
 		if ( loaded && isEditing === null ) {
-			setEditing( ! data.isValid );
+			setEditing( ! isVerified );
+		}
+	}, [ loaded, isVerified, isEditing ] );
+
+	// If `initEditing` is true, EditPhoneNumberCard takes care of the call of `onPhoneNumberVerified`
+	// after the phone number is verified. If `initEditing` is false or null, this useEffect handles
+	// the call of `onPhoneNumberVerified` when the loaded phone number has already been verified.
+	useEffect( () => {
+		if ( loaded && initEditing !== true && isVerified ) {
+			onPhoneNumberVerifiedRef.current();
 		}
-	}, [ loaded, data.isValid, isEditing ] );
+	}, [ loaded, isVerified, initEditing ] );
```

Comment:
❓ Could be possible to have  `isVerified` but not `loaded` phoneNumber?

Reply:
Removed in https://github.com/woocommerce/google-listings-and-ads/pull/1732/commits/290da4181bc70dd5167d709a0a2c73724cc5c45b.